web/blog/unifying-network-and-endpoint-data-with-community-ids/index.md

```diff
@@ -0,0 +1,160 @@
+---
+title: Unifying Network and Endpoint Data with Community IDs
+authors: mavam
+date: 2024-08-29
+tags: [tql, functions, community_id]
+comments: true
+---
+
+The Community ID is a standardized flow hash that generates a unique identifier
+for a network connection. Since it's implemented across multiple tools, it
+allows for tracking events associated with the same connection across different
+systems. This is particularly valuable for correlating network and endpoint
+events.
+
+![Community ID](community-id.svg)
+
+<!-- truncate -->
+
+For instance, when using [Suricata](https://suricata.io) or
+[Zeek](https://zeek.org) for network visibility alongside
+[Sysmon](https://learn.microsoft.com/en-us/sysinternals/downloads/sysmon) or a
+full-fledged EDR for endpoint telemetry, a core challenge is seamlessly linking
+these two types of events to answer questions like:
+
+- Which files were accessed or modified by the process behind this network
+  activity?
+- Which user was logged in when this potentially malicious DNS request was made?
+- What other network connections were established by the same process that
+  communicated with this C2 server?
+- Which process initiated this TLS-encrypted network connection that just
+  matched a JA4 fingerprint?
+- What application is responsible for this anomalous-looking network traffic?
+
+Let's make this more tangible with a concrete example. Consider this Zeek
+connection event:
+
+```json title="conn.log"
+{
+  "ts": "2024-08-29T12:34:56.789Z",
+  "uid": "CZwqhx3td8eTfCSwJb",
+  "id": {
+    "orig_h": "192.168.1.100",
+    "orig_p": 65432,
+    "resp_h": "93.184.216.34",
+    "resp_p": 80
+  },
+  "proto": "tcp",
+  "service": "http",
+  "duration": "5.16s",
+  "orig_bytes": 205,
+  "resp_bytes": 278,
+  "conn_state": "SF",
+  "missed_bytes": 0,
+  "history": "ShADadfF",
+  "orig_pkts": 6,
+  "orig_ip_bytes": 525,
+  "resp_pkts": 5,
+  "resp_ip_bytes": 546,
+  "community_id": "1:7rgwE/NtiMlsxtm1e9OpqJ3icU0="
+}
```

Comment:
```tsv
#separator \x09
#set_separator	,
#empty_field	(empty)
#unset_field	-
#path	conn
#open	2024-08-29-14-05-48.864143
#fields	ts	uid	id.orig_h	id.orig_p	id.resp_h	id.resp_p	proto	service	duration	orig_bytes	resp_bytesconn_state	missed_bytes	history	orig_pkts	orig_ip_bytes	resp_pkts	resp_ip_bytes	community_id
#types	time	string	addr	int	addr	int	string	string	interval	int	int	string	int	string	int	intint	int	string
2024-08-29T12:34:56.789000	CZwqhx3td8eTfCSwJb	192.168.1.100	65432	93.184.216.34	80	tcp	http	5.16s	205	278	SF	0	ShADadfF	6	525	5	546	1:7rgwE/NtiMlsxtm1e9OpqJ3icU0=
```

Reply:
It may make sense to show this as a Zeek TSV log, wdyt?

Reply:
Not a big fan because it's so horizontal and noisy. People not familiar it will not see that data at a glance.

But thanks for the suggestion!